securedrop/manage.py

```diff
@@ -167,11 +169,45 @@ def add_admin():
             break
 
     while True:
+        print "Please enter a password.\n\nThe password must be at least 8 characters and contain a combination of upper\nand lower characters, numbers and special characters.\n"
         password = getpass("Password: ")
         password_again = getpass("Confirm Password: ")
         if password == password_again:
-            break
-        print "Passwords didn't match!"
+            print "Your password is " + str(len(password)) + " characters long."
```

Comment:
All of this code should be factored into a separate, reusable function, to avoid making this function overly long. A good rule of thumb is that any one function should be able to fit on your screen all at once without scrolling. I would've written a function, say `def acceptable_password(password)`, and checked it:

```py
if not acceptable_password(password):
    print "Your password is not strong enough" # TODO explain why, you could do this neatly with exceptions
else:
    break
```